content/en/blog/_posts/XX-userns-enabled-by-default/index.md

```diff
@@ -0,0 +1,266 @@
+[here][userns-k8s-conf]. You can also enable a [feature gate to relax the PSS
+rules][userns-pss].
+
+[userns-k8s-conf]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#set-up-a-node-to-support-user-namespaces
+[task-userns]: https://kubernetes.io/docs/tasks/configure-pod-container/user-namespaces/
+[userns-pss]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#integration-with-pod-security-admission-checks
+
+6. The demos are nice, but are there more CVEs that this mitigates?
+
+Yes, quite a lot, actually! Besides the ones in the demo, the KEP has [more CVEs
+you can check][kep-cve]. That list is not exhaustive, there are many more.
+
+[kep-cve]: https://github.com/kubernetes/enhancements/blob/b8013bfbceb16843686aebbb2ccffce81a6e772d/keps/sig-node/127-user-namespaces/README.md#motivation
+
+7. Can you sum up why user namespaces is important?
+
+Think about running a process as root, maybe even an untrusted process. Do you
+think that is secure? What if we limit it by adding seccomp and apparmor, mask
+some files in /proc (so it can't crash the node, etc.) and some more tweaks?
+
+Wouldn't it be better if we don't give it privileges in the first place, instead
+of trying to play whack-a-mole with all the possible ways root can escape?
+
+This is what user namespaces does, plus some other goodies:
+
+ * **Run as an unprivileged user on the host without making changes to your application**.
+Greg and Vinayak did a great talk on the pains you can face when trying to run
+unprivileged without user namespaces. The pains part [starts in this minute][kubecon-nonroot-pains].
+
+ * **All pods run with different UIDs/GIDs, we significantly improve the lateral
+movement**. This is guaranteed with user-namespaces (the kubelet chooses it for
+you). In the same talk, Greg and Vinayak show how to achieve the same without
+user namespaces we went through a quite complex solution. This part [starts in
+this minute][kubecon-nonroot-uids].
+
+ * **The capabilities granted are only granted inside the user-namespace**. That
+ means that if a pod breaks out of the container, they are not valid on the
+host. We can't provide that without user namespaces.
+
+ * **It enables new use-cases in a _secure_ way**. You can run docker in docker,
+unprivileged container builds, Kubernetes inside Kubernetes, etc all **in a secure
+way**. Most of the previous solutions to do this required privileged containers or
+putting the node at a high risk of compromise.
+
+[kubecon-nonroot-pains]: https://youtu.be/uouH9fsWVIE?feature=shared&t=351
+[kubecon-nonroot-uids]: https://youtu.be/uouH9fsWVIE?feature=shared&t=793
+
+8. Is there container runtime documentation for user namespaces?
+
+Yes, we have:
+ * [containerd
+   documentation](https://github.com/containerd/containerd/tree/b22a302a75d9a7d7955780e54cc5b32de6c8525d/docs/user-namespaces)
+ * CRI-O documentation // TODO: does crio have userns documentation?
```

Comment:
@giuseppe @saschagrunert is there CRIO documentation on userns? I only found about the annotation, that is not what we are talking about here.

Reply:
I am not aware of any, the annotation is a CRI-O specific thing so that was worth documenting, but user namespace support through CRI happens transparently and there is nothing to configure

Reply:
Thanks, I'll just mention that, then :)